firstvoices/backend/tests/test_tasks/test_import_job_tasks.py

```diff
@@ -322,6 +329,39 @@ def test_dry_run_failed(self, caplog):
             assert import_job_instance.validation_status == JobStatus.FAILED
             assert "Random exception." in caplog.text
 
+    def test_failed_rows_csv(self):
+        import_job_instance = self.import_invalid_dictionary_entries()
+        validation_report = import_job_instance.validation_report
+        error_rows = validation_report.rows.all()
+        error_rows_numbers = list(
+            validation_report.rows.order_by("row_number").values_list(
+                "row_number", flat=True
+            )
+        )
+        assert len(error_rows) == 5
+        assert error_rows_numbers == [2, 3, 4, 5, 6]
+
+        # Reading actual file
+        file_content = get_sample_file(
+            "import_job/invalid_dictionary_entries.csv", self.MIMETYPE
+        )
+        input_csv_table = tablib.Dataset().load(
+            file_content.read().decode("utf-8-sig"), format="csv"
+        )
+
+        failed_rows_csv_table = tablib.Dataset().load(
+            import_job_instance.failed_rows_csv.content.read().decode("utf-8-sig"),
+            format="csv",
+        )
+
+        assert len(failed_rows_csv_table) == 5
+
+        for i in range(0, len(error_rows_numbers)):
+            input_index = (
+                error_rows_numbers[i] - 1
+            )  # since we do +1 while generating error row numbers
+            assert input_csv_table[input_index] == failed_rows_csv_table[i]
```

Comment:
This is verifying each row that is returned is the same as the row in the input file without any modifications from us.